### PR TITLE
Bump the fungible token contract

### PR DESCRIPTION
**Pull request by nfurfaro**
based on https://github.com/FuelLabs/fuel-v2-contracts/blob/5f49c4ca866970dd475e04dbe9486fcf260434c3/deployments/deployments.goerli.json#L4

### CODE REVIEW

**Comment by pixelcircuits** [comment]
Looks good! Although a user would need to change several other items in that file too for a correct deployment.

File: bridge-fungible-token/src/bridge_fungible_token.sw [hunk 1]
```diff
@@ -91,13 +91,13 @@ impl BridgeFungibleToken for Contract {
     #[storage(read, write)]
     fn claim_refund(originator: b256, asset: b256) {
         let stored_amount = storage.refund_amounts.get((originator, asset));
```
Comment (pixelcircuits) [inline bridge-fungible-token/src/bridge_fungible_token.sw:93]: Should we just unwrap at this line?

File: bridge-fungible-token/src/utils.sw [hunk 2]
```diff
@@ -173,56 +167,47 @@ pub fn parse_message_data(msg_idx: u8) -> MessageData {
...
+
+    // first, we push the selector 1 byte at a time
+    // the function selector for finalizeWithdrawal on the L1ERC20Gateway contract:
+    // finalizeWithdrawal(address,address,uint256) = 0x53ef1461
+    data.push(0x53u8);
+    data.push(0xefu8);
+    data.push(0x14u8);
+    data.push(0x61u8);
+
+    // next, we copy the least significant 20 bytes of the `to` address:
```
Comment (pixelcircuits) [inline bridge-fungible-token/src/utils.sw:201]: The EVMs ABI encoding expects addresses to be zero padded to a full 32 bytes. We need to copy the whole 32 bytes for both "to" and "LAYER_1_TOKEN".

Comment (nfurfaro) [reply bridge-fungible-token/src/utils.sw:201]: Ah, I see. That's an easy enough change to make.

File: bridge-fungible-token/src/bridge_fungible_token.sw (hunk 1 above)
Comment (nfurfaro) [reply bridge-fungible-token/src/bridge_fungible_token.sw:93]: Yep, makes sense as we then need to use the unwrapped value more than once after this.

**Comment by nfurfaro** [comment]
Note that I'm updating the contract to work with the latest force as well (required to use the stdlib functions and remove the local implementations), and also bumping the SDK from 0..29 => 0.35.

**Review by pixelcircuits — APPROVED** [review]
Nice work!

**Pull request merged**